commit 1577a7bc892c88495d733f3e9889cacdc2755c59 (refs/changes/96/5596/4)
Author: Dan McDonald <danmcd@joyent.com>
Date:   2019-02-19T13:25:17-05:00 (8 months ago)
    
    OS-7586 ipf sometimes freezes RFC 1323 transfers
    Reviewed by: Jason King <jbk@joyent.com>

diff --git a/usr/src/uts/common/inet/ipf/ip_state.c b/usr/src/uts/common/inet/ipf/ip_state.c
index c1fe642d00..184f8775b6 100644
--- a/usr/src/uts/common/inet/ipf/ip_state.c
+++ b/usr/src/uts/common/inet/ipf/ip_state.c
@@ -1729,7 +1729,7 @@ ipstate_t *is;
 		} else if (flags == TH_SYN) {
 			is->is_s0[source] = ntohl(tcp->th_seq) + 1;
 			if ((TCP_OFF(tcp) > (sizeof(tcphdr_t) >> 2)))
-				(void) fr_tcpoptions(fin, tcp, tdata);
+				(void) fr_tcpoptions(fin, tcp, fdata);
 
 			if ((fin->fin_out != 0) && (is->is_pass & FR_NEWISN))
 				fr_checknewisn(fin, is);
@@ -1840,6 +1840,7 @@ int flags;
 	 * the receiver also does window scaling)
 	 */
 	if (!(tcpflags & TH_SYN) && (fdata->td_winflags & TCP_WSCALE_FIRST)) {
+		fdata->td_winflags &= ~TCP_WSCALE_FIRST;
 		fdata->td_maxwin = win;
 	}
 
@@ -1902,7 +1903,7 @@ int flags;
 #endif
 /* XXX what about big packets */
 #define MAXACKWINDOW 66000
-	    (-ackskew <= (MAXACKWINDOW << fdata->td_winscale)) &&
+	    (-ackskew <= (MAXACKWINDOW)) &&
 	    ( ackskew <= (MAXACKWINDOW << fdata->td_winscale))) {
 		inseq = 1;
 	/*
